src/Servers/Kestrel/Core/src/Internal/TlsConnectionFeature.cs

```diff
@@ -14,29 +15,111 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
 {
     internal class TlsConnectionFeature : ITlsConnectionFeature, ITlsApplicationProtocolFeature, ITlsHandshakeFeature
```

Comment:
Making this lazy changes the threading here a little. It means multiple requests will end up writing to these fields lazily. I guess that's not a big deal since these features have settable properties anyways?

Reply:
> It means multiple requests will end up writing to these fields lazily.

? The fields aren't written to in most cases, they're read through. The underlying values should be stable.